unified-runtime/source/adapters/level_zero/device.cpp

```diff
@@ -167,10 +167,9 @@ ur_result_t urDeviceGet(
     std::copy_n(MatchedDevices.begin(), N, Devices);
 
   if (NumDevices) {
-    if (*NumDevices == 0)
-      *NumDevices = ZeDeviceCount;
-    else
-      *NumDevices = N;
+    *NumDevices = ZeDeviceCount;
+  } else {
+    *NumDevices = N;
```

Comment:
Remove this else, this would crash if numDevices is not set since it would dereference the memory.